code/datums/diseases/_disease.dm

```diff
@@ -156,7 +156,7 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 		if(disease_flags & CAN_RESIST)
```

Comment:
That's what is SUPPOSED to happen. CAN_RESIST flag is only suppsoed to add the cured disease to resistances so you are immune to it next time.